### PR TITLE
[Issue-576]fix path error for windows system and optimize getPath method

### PR DESCRIPTION
**Pull request by QiangCai**
**1**  use FileUtils.getPaths method  to get input files for dataload.
 invoke listStatus only one time per dataloading.

**2** optimize FileUtils.getPaths to skip input files as following.
   a: file is empty
   b: file path start with "_"  or "."
   c: file path doesn't end with  ".csv"

Link to issue:  [Issue-576 ](https://github.com/HuaweiBigData/carbondata/issues/576)

### CODE REVIEW

**Comment by QiangCai** [comment]
@Jay357089 please review this pr

**Comment by Zhangshunyu** [comment]
test ok on the windows platform.

File: integration/spark/src/main/scala/org/apache/spark/util/FileUtils.scala [hunk 1]
```diff
@@ -17,26 +17,33 @@
...
+    case false =>
+      val path = carbonFile.getPath
+      if (carbonFile.getSize == 0) {
+        logWarning(s"skip empty input file: $path")
+      } else if (path.startsWith("_") || path.startsWith(".")) {
+        logWarning(s"skip invisible input file: $path")
+      } else if (path.toLowerCase().endsWith(".csv")) {
+        stringBuild.append(path.replace('\\', '/')).append(",")
+      } else {
+        logWarning(s"skip input file: $path, beacase this path don't end with '.csv'")
```
Comment (Jay357089) [inline integration/spark/src/main/scala/org/apache/spark/util/FileUtils.scala]: @QiangCai because ,  doesn't

Comment (QiangCai) [inline integration/spark/src/main/scala/org/apache/spark/util/FileUtils.scala]: fixed

**Comment by QiangCai** [comment]
@ravipesala please review and merge code

**Comment by kumarvishal09** [comment]
retest this please

**Comment by QiangCai** [comment]
@kumarvishal09  please review

File: integration/spark/src/main/java/org/carbondata/spark/util/CarbonQueryUtil.java [hunk 2]
```diff
@@ -153,10 +154,23 @@ public static void getAllFiles(String sourcePath, List<String> partitionsFiles,
     }
   }
 
+  /**
+   * split sourcePath by comma
+   */
+  public static void splitFilePath(String sourcePath, List<String> partitionsFiles,
+      String separator) {
+    if (StringUtils.isNotEmpty(sourcePath)) {
+      String[] files = sourcePath.split(",");
```
Comment (sujith71955) [inline integration/spark/src/main/java/org/carbondata/spark/util/CarbonQueryUtil.java]: For ',' separator please define a constants variable or reuse the existing one.

File: integration/spark/src/main/scala/org/apache/spark/util/FileUtils.scala [hunk 3]
```diff
@@ -46,16 +53,21 @@ object FileUtils {
...
       for (i <- 0 until filePaths.size) {
         val fileType = FileFactory.getFileType(filePaths(i))
         val carbonFile = FileFactory.getCarbonFile(filePaths(i), fileType)
-        stringbuild.append(getPathsFromCarbonFile(carbonFile)).append(",")
+        getPathsFromCarbonFile(carbonFile, stringBuild)
+      }
+      if (stringBuild.nonEmpty) {
+        stringBuild.substring(0, stringBuild.size - 1)
+      } else {
+        throw new IllegalArgumentException("please check your input path and make sure " +
```
Comment (sujith71955) [inline integration/spark/src/main/scala/org/apache/spark/util/FileUtils.scala]: Dont throw raw exception, please throw module specific exception, Please check

File: integration/spark/src/main/java/org/carbondata/spark/util/CarbonQueryUtil.java (hunk 2 above)
Comment (QiangCai) [inline integration/spark/src/main/java/org/carbondata/spark/util/CarbonQueryUtil.java]: fixed

File: integration/spark/src/main/scala/org/apache/spark/util/FileUtils.scala (hunk 3 above)
Comment (QiangCai) [inline integration/spark/src/main/scala/org/apache/spark/util/FileUtils.scala]: fixed

**Comment by QiangCai** [comment]
@sujith71955 done

**Comment by sujith71955** [comment]
retest this please

**Comment by QiangCai** [comment]
retest this please

**Comment by sujith71955** [comment]
LGTM

**Pull request merged**